rfcs/2021-11-22-design-tokens.md

```diff
@@ -0,0 +1,106 @@
+
+   I believe the best option would be to write our own [Webpack Loader](https://webpack.js.org/contribute/writing-a-loader/) to convert these tokens to top level Sass files.
+
+   1. In the case of the DS-Core, we would convert these tokens into variables which would be utilized by files in our /src/styles/settings/variables scss files and font.scss file initially. Since we would not be changing these files aside from referencing the token variable names in the tokens NPM package instead of providing hard coded values, the compiled output of these files should be compared to detect any breaking changes. One advantage to using a custom webpack loader in our case is that we could do things like generate the list of @font-face includes based on what tokens are available, slimming overall DS package size.
+   2. In the case of Sketch and the DSM, the tool chosen/created would be extended to output libraries for the DSM (if needed). Sketch and these libraries would also be versioned and live in the token package. These would be made available as release download libraries with which to work from. The [Sketch file format](https://developer.sketch.com/file-format/) is very easy to work with, as it is already made up of JSON files.
+
+4. Future work to structure similarities between component level variables in Sketch and the codebase would be ongoing from this point. These variables could be added and utilized on a component-by-component basis, and would make use of the highest level tokens for spacing, color, etc.. i.e.
+
+   ```
+   button-small-active-highlight-inverse-color (sketch)
+   $button-small-active-highlight-inverse-color (scss)
+
+   tokens.git/core/components/buttons.json
+   {
+   	"button": {
+   		"small": {
+   			"active": {
+   				"highlight": {
+   					"color": "ds.core.colors.lightblue.300"
+   				}
+   			}
+   		}
+   	}
+   }
+   ```
+
+5. Documentation would need to be generated from the token build task. This could be another option in our existing site docs, filed under "Design Tokens." Component level variables could be included on the doc pages for each component.
+
+### Other Notes
+
+- There are several ways to transform values from one format to another. If we agree on standard units (`px`, `rem`, `rgb`) for each token type, we can utilize Sass Mixins to transform these values to other types dynamically. [A CodePen example](https://codepen.io/jakob-e/pen/AHunv).
+- There are many tickets currently in the backlog that would be resolved with this work automatically.
+
+### Questions and Requested Feedback
```

Comment:
I'm curious about utilize a token set from another design system like USWDS? https://designsystem.digital.gov/design-tokens/ 

Is there a path where we use their token set?

Reply:
It seems like that would be tough to work out, it would be a larger shift. That would require a large shift at many levels in our system, what I'm proposing can be done incrementally with smaller changes and deprecations along the way.